plugins/catalog-graphql/src/schema.gql

```diff
@@ -0,0 +1,25 @@
+type EntityMetadataAnnotation {
+  key: String!
+  value: String!
+}
+
+type EntityMetadata {
+  name: String!
+  namespace: String
+  annotation(name: String): EntityMetadataAnnotation
+  annotations: [EntityMetadataAnnotation!]
+}
+
+type CatalogEntity {
+  apiVersion: String
+  kind: String!
+  metadata: EntityMetadata
```

Comment:
could add `spec: JSONObject` too if we add that

Reply:
I think we would lose the typing on it then. We could do something like `rawSpec` if we wanted to do that, but think it's easier with the functionality in #1867